### PR TITLE
`get_logit_fn_tables` and other support functions

### PR DESCRIPTION
**Pull request by bpbond**
These functions are needed for `L203.demand_input` ( #452 ). Basically straight translations of the originals, although I've tweaked parameters names and ordering, and provided assertions and test code.

@rynacui - please review.

@pkyle - I understand technically what's going on in these, but not always what the rationale is, and so the documentation for all these functions needs improvement so that it clearly states _why_ they do what they do. Can you look through and, either in comments or by pushing to this PR, help? Thanks.

### CODE REVIEW

**Comment by bpbond** [comment]
Thanks @rynacui . Since you're holding on these functions, I'm going to go ahead and merge, and open an issue for @pkyle and/or me to bring their documentation up to par later.

File: R/module-helpers.R [hunk 1]
```diff
@@ -85,3 +85,171 @@ rename_SO2 <- function(x, so2_map, is_awb = FALSE) {
...
+  }
+
+  # Loop through each of the logit types and create a table for it
+  # using the appropriate header name and writing to all regions if requested.
+  for(curr_logit_type in gcam.LOGIT_TYPES) {
+    tables[[curr_logit_type]]$header <- paste0(base_header, curr_logit_type)
+    currdata <- data[data$logit.type == curr_logit_type,]
+    if(write_all_regions) {
+      if(nrow(currdata) > 0) {
+        currdata <- write_to_all_regions(currdata, names, ...)
```
Comment (rynacui) [inline R/module-helpers.R]: `write_to_all_regions` needs to specify `GCAM_region_names`. When use separately, `write_to_all_regions(data, names, GCAM_region_names = GCAM_region_names)`, first is the parameter and second is the tibble. But here the tibble is not specified, generates error: `argument "GCAM_region_names" is missing, with no default`.

Comment (bpbond) [inline R/module-helpers.R]: Ah, sorry. I'll fix.

Comment (bpbond) [inline R/module-helpers.R]: Opened #467

**Pull request merged**